data/campaigns/Eastern_Invasion/utils/traits.cfg

```diff
@@ -0,0 +1,312 @@
+        id=audacious
+        male_name= _ "audacious"
+        description= _ "Fights normally during unfavorable times of day/night"
+        help_text= _ "Aversion to light and dark holds no sway over these audacious individuals."
+        [effect]
+            apply_to="fearless"
+        [/effect]
+    [/trait]
+#enddef
+
+#define TRAIT_DESERTER
+    [trait]
+        id=deserter
+        male_name= _ "deserter"
+        female_name= _ "female^deserter"
+        description= _ "When this unit drops below half health, it will flee the battle and reappear on your recall list."
+        # NOTE - implmentation is handled using [event]s in _main.cfg
+    [/trait]
+#enddef
+
+#define DESERTER_EVENT FLT UNIT
+    first_time_only=no
+    [{FLT}]
+        side=1 # don't use this trait for the AI humans in S20
+        trait=deserter
+        formula=(self.hitpoints < self.max_hitpoints/2)
+    [/{FLT}]
+    [filter_condition]
+        [have_unit]
+            side,id=1,Dacyn # prevent this from triggering at the very beginning of S11_Captured, when fighting bats (since you can't desert when you're already a prisoner)
+        [/have_unit]        # also prevents this from triggering in S20, when the deserters are AI-controlled
+    [/filter_condition]
+
+    {RANDOM (_"Forget this!",_"Run for your lives!",_"Get me out of here!")}
+    [message]
+        speaker=${UNIT}.id
+        message=$random
+    [/message]
+    {CLEAR_VARIABLE random}
+    [put_to_recall_list]
+        id,heal=${UNIT}.id,yes
+    [/put_to_recall_list]
+    [message]
+        speaker=narrator
+        message= _ "The ‘deserter’ trait has caused ${UNIT}.name to flee back to your recall list."
+        image=wesnoth-icon.png
+    [/message]
+#enddef
+#define GLOBAL__TRAIT_DESERTER
+    [event]
+        name=attacker hits
+        {DESERTER_EVENT filter_second second_unit}
+    [/event]
+    [event]
+        name=defender hits
+        {DESERTER_EVENT filter unit}
+    [/event]
+#enddef
+
+#define TRAIT_MERCENARY
+    [trait]
+        id=mercenary
+        male_name= _ "mercenary"
+        female_name= _ "female^deserter"
+        description= _ "<span color='#00FF00'>+30%</span> base damage when gold is positive.
+<span color='#FF0000'>-30%</span> base damage when gold is 0 or negative."
+    [/trait]
+#enddef
+
+#define GLOBAL__TRAIT_MERCENARY
+    # only supported for side 1
+    [event]
+        name=side 1 turn refresh # after this turn's income
+        first_time_only=no
+        [fire_event]
+            name=mercenary_refresh
+        [/fire_event]
+    [/event]
+    [event]
+        name=recruit,recall,moveto # these events can cost or gain gold
+        first_time_only=no
+
+        [filter]
+            side=1
+        [/filter]
+
+        [fire_event]
+            name=mercenary_refresh
+        [/fire_event]
+        [on_undo]
+            [fire_event]
+                name=mercenary_refresh
+            [/fire_event]
+        [/on_undo]
```

Comment:
i think this has some risk to cause OOS (or replay errors) in rare corner cases. The problem is that first removing an object and then reapplying it is not really a nop since it might change the order or objects if an another object after the buff object was added, if it was for example an object  that increases damage by fixed amount.

It might be better to implement this as a [damage] ability with a filter that checks the sides gold (i _think_ this is possible in formula but not sure)

Reply:
> It might be better to implement this as a [damage] ability with a filter that checks the sides gold (i _think_ this is possible in formula but not sure)

Assuming a StandardUnitFilter, I think `[filter_side]formula="gold ..."` would do it?

Reply:
Can you show the code that you tried ?

Reply:
`#define TRAIT_MERCENARY
    [trait]
        id=mercenary
        male_name= _ "mercenary"
        female_name= _ "female^deserter"
        description= _ "<span color='#00FF00'>+30%</span> base damage when gold is positive.
<span color='#FF0000'>-30%</span> base damage when gold is 0 or negative."

        [effect]
            [filter]
                [filter_side]
                    formula="side_number.gold>0"
                [/filter_side]
            [/filter]
            apply_to,increase_damage=attack,+30%
        [/effect]
        [effect]
            [filter]
                [filter_side]
                    formula="side_number.gold<=0"
                [/filter_side]
            [/filter]
            apply_to,increase_damage=attack,-30%
        [/effect]
    [/trait]
#enddef`

Reply:
I tested the suggestion myself and it seemed to work 
```

	    	[trait]
	    	    id=mercenary
	    	    male_name= _ "mercenary"
	    	    female_name= _ "female^deserter"
	    	    description= _ "<span color='#00FF00'>+30%</span> base damage when gold is positive.
<span color='#FF0000'>-30%</span> base damage when gold is 0 or negative."
				[effect]
        		    apply_to=new_ability
    		        [abilities]
						[damage]
							[filter]
								[filter_side]
									formula="gold > 0"
								[/filter_side]
							[/filter]
							multiply=1.3
						[/damage]
						[damage]
							[filter]
								[filter_side]
									formula="gold < 0"
								[/filter_side]
							[/filter]
							multiply=0.7
						[/damage]
		            [/abilities]
				[/effect]
    		[/trait]
```

Reply:
Oh nice, it works. Thanks!